docs/hooks.md

```diff
@@ -0,0 +1,127 @@
+# Hooks
+
+UIx wraps existing React hooks to cut some corners and provide a more idiomatic interafce for Clojure. `uix.core` exposes only default React hooks, named against their JS counterparts but in kebab-case.
+
+There are multiple differences from pure React though.
+
+## Dependency array
+
+Some hooks accept an array of dependencies as the second argument. While in pure React it has to be an array literal `#js []`, in UIx as Clojure library the deps are a vector literal `[]`.
+
+```clojure
+(uix/use-effect
+  (fn [] (prn x))
+  [x])
+```
+
+## How depedencies are compared?
+
+When the same depency has a different value between component updates, a hook will re-run.
+
+```clojure
+;; 1st update
+(uix/use-effect
+  (fn [] (prn x)) ;; prn 1
+  [x]) ;; x = 1
+
+;; 2nd update
+(uix/use-effect
+  (fn [] (prn x))
+  [x]) ;; x = 1, didn't change, do nothing
+
+;; 3rd update
+(uix/use-effect
+  (fn [] (prn x)) ;; prn 2
+  [x]) ;; x = 2, `x` has changed, re-run the hook
+```
+
+That works as expected for primitive values like numbers and strings, but what about Clojure's maps and vectors that can be compared by value `(= {:x 1} {:x 1}) ;; true`?
+
+Two maps equal by value but with different identity are considered to be not equal when React compares dependencies. This is happening because React is JS library that is not aware of Clojure's equality algorithm, instead React is comparing values by identity `===` or `Object.is`. Thus it's important to think about what type of values you are passing as dependencies of a hook. It applies to JS as well, objects and arrays are still compared by reference.
+
+```clojure
+;; 1st update
+(let [a {:x 1}]
+  (uix/use-effect
+    (fn [] (prn a)) ;; executed
+    [a])))
+
+;; 2nd update
+(let [a {:x 1}]
+  (uix/use-effect
+    (fn [] (prn a)) ;; executed as well, since `a` is a new map that was created during render
+    [a])))
+
+;; 3rd update
+(let [a {:x 1}]
+  (uix/use-effect
+    (fn [] (prn a)) ;; same, still executed for the same reason
+    [a])))
+```
+
+### What about other Clojure's primitives?
+
+In addition to JavaScript's primitive types like `Number` or `String`, ClojureScript has keywords, symbols and UUIDs that are represented as JS objects when compiled into JS and thus fall into the same trap with equality check. To make things simpler UIx automatically stringifies those three types when passed in dependency vector.
+
+## Return value in effect hooks
+
+The _setup_ function that is passed into one of effect hooks requires the return value to be either a function (that will be called on _cleanup_) or `js/undefined`. Otherwise React will throw an error saying that it got something else.
```

Comment:
```suggestion
The _setup_ function, that is passed into one of the effect hooks, requires the return value to be either a function (that will be called on _cleanup_) or `js/undefined`. Otherwise React will throw an error saying that it got something else.
```